### PR TITLE
Add relative_ast_path_feature

### PR DESCRIPTION
**Pull request by keith**
This feature applies for darwin builds to disable absolutizing
swiftmodule paths to link commands. Previously these paths were
absolutized to support debugging. This is not required if users change
the pwd of their debugger so that the relative paths are valid. Since we
don't have a way to force that on users, absolutizing them is still the
default, but users who configure that can enable this feature to improve
hermeticity of their builds.

RELNOTES: Add `relative_ast_path` feature for darwin builds to relativize swiftmodule paths for debugging

### CODE REVIEW

**Comment by keith** [comment]
tbh I'm not sure a feature / env var is the best way to handle this, but I can't think of another way to have a "global" setting like this

File: tools/osx/crosstool/cc_toolchain_config.bzl [hunk 1]
```diff
@@ -5218,6 +5218,24 @@ def _impl(ctx):
         ],
     )
 
+    relative_ast_path_feature = feature(
+        name = "relative_ast_path",
+        env_sets = [
+            env_set(
+                actions = all_link_actions + [
+                    "objc-executable",
+                    "objc++-executable",
```
Comment (thii) [inline tools/osx/crosstool/cc_toolchain_config.bzl:5227]:
```suggestion
                    ACTION_NAMES.objc_executable,
                    ACTION_NAMES.objcpp_executable,
```

**Comment by keith** [comment]
@thii can you accept the CLA

**Comment by thii** [comment]
@googlebot I fixed it.

**Comment by keith** [comment]
@oquenchil I would love this to make the 3.8 cut if possible

**Review by oquenchil — APPROVED** [review]
LGTM

**Comment by keith** [comment]
Thanks! Can you help land this or do we need to cc someone?

**Comment by keith** [comment]
Thanks!

**Pull request closed without merging**